tools/lsp/language/completion.rs

```diff
@@ -665,46 +665,31 @@ pub fn build_import_statements_edits(
         Position::new(map_position(&token.source_file, last.into()).line + 1, 0)
     };
 
-    for file in document_cache.documents.all_files() {
-        let Some(doc) = document_cache.documents.get_document(file) else { continue };
-        let file = if file.starts_with("builtin:/") {
-            match file.file_name() {
-                Some(file) if file == "std-widgets.slint" => "std-widgets.slint".into(),
-                _ => continue,
-            }
-        } else {
-            match lsp_types::Url::make_relative(
-                &current_uri,
-                &lsp_types::Url::from_file_path(file)
-                    .unwrap_or_else(|()| panic!("Cannot parse URL for file '{file:?}'")),
-            ) {
-                Some(file) => file,
-                None => continue,
-            }
+    let exports = {
+        let mut tmp = Vec::new();
+        document_cache.all_exported_components(&mut tmp);
```

Comment:
this `all_exported_component` function is quite expensive though.  it allocates multiple times per component.
The previous code tries to limit the amount of allocations that's why there is a `filter` function instead of doing nothing in `add_edit`. But maybe that was over-engineered?  but we could pass the filter function to all_exported_component, or use the same pattern with `filter` and `add_component`

Reply:
But constructing the ComponentInformation in is the most expensive as it allocates several time for various strings. Anyway. It's probably fine like that. No need to be the most optimal in that function.